src/features/access-package/edit/changed_accesses.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import List from "@material-ui/core/List";
+import ListItem from "@material-ui/core/ListItem";
+import ListItemText from "@material-ui/core/ListItemText";
+import {Typography} from "@material-ui/core";
+import {makeStyles} from "@material-ui/core/styles";
+import ListItemIcon from "@material-ui/core/ListItemIcon";
+import AddIcon from "@material-ui/icons/AddCircleRounded";
+import RemoveIcon from "@material-ui/icons/RemoveCircleRounded";
+import {useSelector} from "react-redux";
+
+
+const useStyles = makeStyles(theme => ({
+    addingText: {
+        color: "green",
+    },
+    removingText: {
+        color: "red",
+    }
+}));
+
+const ChangedAccesses = (props) => {
+    const classes = useStyles();
+    const {oldAccessPackage, newAccessPackage} = props;
+    const componentConfiguration = useSelector(state => state.component_configuration.componentConfiguration);
+    const collectionList = [];
+    const readList = [];
+    const modifyList = [];
+
+    function accessListItem(entry) {
+        const removingEntry = entry.charAt(0) === "-";
+        let entity = entry.substring(2, entry.length);
+        componentConfiguration.map(componentConfiguration => {
+            componentConfiguration.classes.map(aClass => {
+                if (aClass.path === entity){
+                    entity = componentConfiguration.displayName + " " + aClass.name;
+                }
+                return null;
+            })
+            return null;
+        });
+        return (
+            <ListItem key={entry}>
+                <ListItemIcon>
+                    {removingEntry ? <RemoveIcon className={classes.removingText}/> :
+                        <AddIcon className={classes.addingText}/>}
+                </ListItemIcon>
+                <ListItemText
+                    primary={entity}
+                />
+            </ListItem>
+        );
+    }
+
+    return (
+        <div>
+            <List dense>
```

Comment:
break this up in methods to make it more readable.

Reply:
Extracted to method